it/src/main/java/com/google/cloud/teleport/it/datastream/DefaultDatastreamResourceManager.java

```diff
@@ -0,0 +1,306 @@
+            connectionProfileBuilder.setMysqlProfile(mysqlProfileBuilder);
+        } else if (this.source == Source.ORACLE) {
+            oracleProfileBuilder.setHostname(hostname).setUsername(username).setPassword(password).setPort(1521);
+            connectionProfileBuilder.setOracleProfile(oracleProfileBuilder);
+        } else if (this.source == Source.ORACLE) {
+            postgresqlProfileBuilder.setHostname(hostname).setUsername(username).setPassword(password)
+                    .setPort(5432).setDatabase(postgresDb);
+            connectionProfileBuilder.setPostgresqlProfile(postgresqlProfileBuilder);
+        }
+
+        ConnectionProfile connectionProfile = connectionProfileBuilder
+                .setDisplayName(connectionProfileId)
+                .setStaticServiceIpConnectivity(StaticServiceIpConnectivity.getDefaultInstance())
+                .build();
+        CreateConnectionProfileRequest request = CreateConnectionProfileRequest.newBuilder()
+                .setParent(LocationName.of(projectId, location).toString())
+                .setConnectionProfile(connectionProfile)
+                .setConnectionProfileId(connectionProfileId)
+                .build();
+        ConnectionProfile reference = datastreamClient.createConnectionProfileAsync(request).get();
+        createdConnectionProfileIds.add(connectionProfileId);
+
+        return reference;
+    }
+
+    @Override
+    public ConnectionProfile createDestinationConnectionProfile(String connectionProfileId)
+            throws ExecutionException, InterruptedException {
+        ConnectionProfile.Builder connectionProfileBuilder = ConnectionProfile.newBuilder()
+                .setDisplayName(connectionProfileId)
+                .setStaticServiceIpConnectivity(StaticServiceIpConnectivity.getDefaultInstance());
+
+        if (this.destination == Destination.GCS) {
+            connectionProfileBuilder
+                    .setGcsProfile(GcsProfile.newBuilder().setBucket(this.gcsBucketName).setRootPath(this.gcsRootPath));
+        }
+        // TODO: BQ profile builder
+
+        CreateConnectionProfileRequest request = CreateConnectionProfileRequest.newBuilder()
+                .setParent(LocationName.of(projectId, location).toString())
+                .setConnectionProfile(connectionProfileBuilder)
+                .setConnectionProfileId(connectionProfileId)
+                .build();
+
+        ConnectionProfile reference = datastreamClient.createConnectionProfileAsync(request).get();
+        createdConnectionProfileIds.add(connectionProfileId);
+
+        return reference;
+    }
+
+    public SourceConfig buildSourceConfig(String sourceConnectionProfileId) {
+        SourceConfig.Builder sourceConfigBuilder = SourceConfig
+                .newBuilder()
+                .setSourceConnectionProfile(ConnectionProfileName
+                        .format(projectId, location, sourceConnectionProfileId));
+
+        if (this.source == Source.MYSQL) {
```

Comment:
This can also be changed to a switch statement like my previous suggestion.

Reply:
done